internal/api/core/server.go

```diff
@@ -65,25 +66,68 @@ func NewPersesAPI(serviceManager dependency.ServiceManager, persistenceManager d
 		configendpoint.New(cfg),
 		migrateendpoint.New(serviceManager.GetMigration()),
 		validateendpoint.New(serviceManager.GetSchemas(), serviceManager.GetDashboard()),
-		authendpoint.New(persistenceManager.GetUser(), serviceManager.GetCrypto()),
+		authendpoint.New(persistenceManager.GetUser(), serviceManager.GetJWT()),
 	}
 	return &api{
 		apiV1Endpoints: apiV1Endpoints,
 		apiEndpoints:   apiEndpoints,
+		jwtMiddleware: serviceManager.GetJWT().Middleware(func(c echo.Context) bool {
+			return !*cfg.ActivatePermission
+		}),
 	}
 }
 
 func (a *api) RegisterRoute(e *echo.Echo) {
-	a.registerAPIV1Route(e)
+	// First, let's collect every route.
+	// The expecting result is a tree we will need to loop over.
+	groups := a.collectRoutes()
+	// Now let's create a simple struct that will help us to loop over the route tree.
+	type queue struct {
+		parent *echo.Group
+		group  *shared.Group
+	}
+	var queueList []queue
```

Comment:
```suggestion
	queueList := make([]queue, 0, len(groups))
```

Reply:
The maximum size of `queueList` can be more than the length of `groups`. Not sure if it really worth to initialise the size of `queueList` in that case

Reply:
Oh nvm!
I see we might append to the queueList on the second loop.